build/org.eclipse.pde.build/src/org/eclipse/pde/internal/build/ProductGenerator.java

```diff
@@ -176,7 +177,7 @@ public boolean generateP2Info() throws CoreException {
 		if (initialInf.exists()) {
 			//copy over initial contents
 			try {
-				StringBuffer inf = Utils.readFile(initialInf);
+				String inf = Files.readString(initialInf.toPath(), Charset.defaultCharset());
```

Comment:
When explicitly using a charset it should be thought of whether that charset is really the right one to use. using no charset at least gives a static code analysis warning. In this case my best guess is that it should be a fixed, ASCII compatible charset. or even only 7 bit Ascii if it's not agreed to use UTF-8.

Reply:
OK. I'll try to find out what charset is used to write those files.

But for me, passing no charset does not produce a warning for me, at least with Sonar-Lint.

Reply:
Looks like this file is only written with `Files.writeString()` below in this method.
That being said, the file is also loaded as Properties file, so it should be `ISO_8859_1` encoded.